compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFinder.java

```diff
@@ -1732,6 +1750,41 @@ public void visit(BLangReFlagExpression reFlagExpression) {
         lookupNode(reFlagExpression.colon);
     }
 
+    private void lookupResourceAccessPathSegments(BLangInvocation.BLangResourceAccessInvocation resourceInvocation) {
+        if (resourceInvocation.targetResourceFunc == null) {
+            // Return if target-function is not set.
+            // Ex: Ambiguous target-functions
+            return;
+        }
+
+        List<BResourcePathSegmentSymbol> pathSegSymbols = resourceInvocation.targetResourceFunc.pathSegmentSymbols;
+        List<BLangExpression> pathExprs = resourceInvocation.resourceAccessPathSegments.exprs;
+
+        if (pathExprs.size() == 0 && pathSegSymbols.get(0).getKind() == SymbolKind.RESOURCE_ROOT_PATH_SEGMENT) {
```

Comment:
Are we certain that the  pathSegSymbols list is not empty at this point?

Reply:
Yeah. For any resource function, there should be any paths or dot resource paths. Those functions that don't have a dot or any path should fail before coming here